c_src/dpiData_nif.c

```diff
@@ -365,40 +347,25 @@ DPI_NIF_FUN(data_get)
         break;
     case DPI_NATIVE_TYPE_STMT:
     {
-        dpiStmt_res *stmtRes = NULL;
-        if (!dataRes->stmtRes)
+        dpiStmt_res *stmtRes = (dpiStmt_res *)dataRes->stmtRes;
+        if (!stmtRes)
         {
             // first time
             stmtRes = enif_alloc_resource(dpiStmt_type, sizeof(dpiStmt_res));
-            stmtRes->stmt = data->value.asStmt;
             dataRes->stmtRes = stmtRes;
         }
-        else
-        {
-            // possible reuse attempt
-            stmtRes = (dpiStmt_res *)dataRes->stmtRes;
-            if (stmtRes->stmt != data->value.asStmt)
-            {
-                // ref cursor changed
-                enif_release_resource(stmtRes);
```

Comment:
will this not create some leaks or the reference never changes ?

Reply:
previously we had
```c
enif_release_resource(stmtRes);
stmtRes = enif_alloc_resource(...);
```
which was deteling the old resource and creating a new one just to change the `dpiStmt *` (`stmtRes->stmt = data->value.asStmt;`)

Instead of this re-creation, here I am re-using the existing resource while changing the `dpiStmt *stmt` member value within it.
https://github.com/K2InformaticsGmbH/oranif/blob/2957af73b4f5768d9fe961133c8776d57c368dff/c_src/dpiStmt_nif.h#L12-L16

So effectively now the reference isn't changing only data within it

Reply:
Got it, thanks for the clarification